examples/OfflineLogin.md

```diff
@@ -0,0 +1,90 @@
+
+#### GetInstanceAsync
+
+This method will complete after the realm has been opened and is fully synchronized. For this reason, this method is recommended to use only when it is essential to work with a fully synchronized realm, for example just after the first login of the user on a freshly installed application. 
+
+The caveat with this method, though, is that it will continue to retry connecting to the server even if the device is offline, and could potentially run forever. In order to avoid this, you have two main possibilities:
+- Use a cancellation token. The `GetInstanceAsync` method accept an optional `CancellationToken` variable as input, that can be used to cancel the operation after a maximum delay, like in the following example:
+    ```csharp
+    try
+    {
+        var cts = new CancellationTokenSource(4000);
+        var configuration = new FlexibleSyncConfiguration(app.CurrentUser);
+        var realm = await Realm.GetInstanceAsync(configuration, cts.Token);
+    }
+    catch (TaskCanceledException tce)
+    {
+        Console.WriteLine("GetInstanceAsync timeout");
+    }
+    ```
+    This is the recommended approach in most of the cases, as it allows to configure the most appropriate timeout for the operation, depending on the specific application scenario.
+- Set `CancelAsyncOperationsOnNonFatalErrors` to `true` on `FlexibleSyncConfiguration`. When set to `true`, async operations such as `Realm.GetInstanceAsync` or `Session.WaitForUploadAsync` will throw an exception whenever a non-fatal error, such as a timeout occurs. The timeouts can be customized by modifying the values in `AppConfiguration.SyncTimeoutOptions`, that contains a series of properties that control sync timeouts, such as the connection timeout and others. For example:
+    ```csharp
+    var appConfiguration = new AppConfiguration(config.AppId);
+    appConfiguration.SyncTimeoutOptions.ConnectTimeout = TimeSpan.FromSeconds(4);
+    app = Realms.Sync.App.Create(appConfiguration);
+
+    // ....
+
+    try
+    {
+        var configuration = new FlexibleSyncConfiguration(app.CurrentUser)
+        {
+            CancelAsyncOperationsOnNonFatalErrors = true
+        };
+        var realm = await Realm.GetInstanceAsync(configuration);
+    }
+    catch (Exception ex)
+    {
+        Console.WriteLine("GetInstanceAsync timeout");
+    }
+    ```
+    In most cases this approach could be an overkill, as the exception would be raised for any kind of non-fatal errors, even for just a brief connection issue. For this reason it is recommended to use the previous approach, as it allows to retry in the specified time frame.
+
+### Recommended flow
+
+Given the differences between `GetInstance` and `GetInstanceAsync` that we have presented, here is an example of a recommended flow that uses both methods to open a realm, and that can be used in most applications:
+
+```csharp
+var user = app.CurrentUser;
+var realmConfig = GetRealmConfig();
```

Comment:
You can't create a config without the user, so this will need to be moved further down.